hcipy/plotting/animation.py

```diff
@@ -1,43 +1,77 @@
-import matplotlib.pyplot as plt
+import glob
+import os
+from subprocess import Popen, PIPE
+
 import matplotlib
-from subprocess import Popen, PIPE, call
 import imageio
-import os
+from PIL import Image
+
 
 class GifWriter(object):
 	def __init__(self, filename, framerate=15):
 		self.closed = False
 		self.filename = filename
 		self.framerate = framerate
 
-		if not os.path.exists(filename + '_frames/'):
-			os.mkdir(filename + '_frames/')
-		self.frame_number = 0
+		self.path_to_frames = self.filename + "_frames"
+		if not os.path.exists(self.path_to_frames):
+			os.mkdir(self.path_to_frames)
+		self.n_frames = 0
 	
 	def add_frame(self, fig=None, arr=None, cmap=None):
 		if self.closed:
 			raise RuntimeError('Attempted to add a frame to a closed GifWriter.')
-		
-		dest = self.filename + '_frames/%05d.png' % self.frame_number
-		self.frame_number += 1
+
+		dest = os.path.join(self.path_to_frames, '%05d.png' % self.n_frames)
 
 		if arr is None:
 			if fig is None:
-				fig = plt.gcf()
+				fig = matplotlib.pyplot.plt.gcf()
 			fig.savefig(dest, format='png', transparent=False)
 		else:
 			if not cmap is None:
 				arr = matplotlib.cm.get_cmap(cmap)(arr, bytes=True)
 			
 			imageio.imwrite(dest, arr, format='png')
-	
+
+		self.n_frames += 1
+
+	@staticmethod
+	def convert2gif(dest_filename, src_file_path, framerate,  src_file_suffix="png", n_files2convert=None):
+		search_pattern = os.path.join(src_file_path, "*."+src_file_suffix)
+		files = glob.glob(search_pattern)
+		files.sort()
+
+		if n_files2convert is not None and len(files) != n_files2convert:
+			raise OSError("Expected {} files but found {}".format(n_files2convert, len(files)))
+
+		# Open all frames to convert
+		frames = []
+		for file in files:
+			frames.append(Image.open(file))
+
+		# Convert to GIF
+		# https://pillow.readthedocs.io/en/stable/handbook/image-file-formats.html?highlight=duration#saving
+		# duration := display duration of each frame in ms
+		duration = int(1000 / framerate)
+		frames[0].save(dest_filename,
+						format="GIF",
+						append_images=frames[1:],
+						save_all=True,
+						duration=duration,
+						loop=0)
+
+	def convert(self):
+		self.convert2gif(self.filename, self.path_to_frames, self.framerate, n_files2convert=self.n_frames)
+
 	def close(self):
-		if not self.closed:
-			command = ['convert', '-delay', str(int(100/self.framerate)), 
-						'-loop', '0', self.filename + '_frames/*.png', self.filename]
-			call(command)	
-			call(['rm', '-rf', self.filename + '_frames'])
```

Comment:
I removed this but am happy to put back if you want it to remain.

We'll want to not have it do this but I can just inherit and override.

Reply:
Make it an option in `__init__()` called `cleanup` with default `True`.